lib/common_test/doc/src/ct_hooks_chapter.xml

```diff
@@ -380,111 +384,85 @@
-     {Config, State}.
-
  %% Called after end_per_suite.
- post_end_per_suite(Suite,Config,Return,State) ->
+ post_end_per_suite(Suite,_Config,Return,State) ->
      Data = {suites, Suite, State#state.suite_total, lists:reverse(State#state.tcs)},
      {Return, State#state{ data = [Data | State#state.data] ,
                            total = State#state.total + State#state.suite_total } }.
 
- %% Called before each init_per_group.
- pre_init_per_group(Suite,Group,Config,State) ->
-     {Config, State}.
-
- %% Called after each init_per_group.
- post_init_per_group(Suite,Group,Config,Return,State) ->
-     {Return, State}.
-
- %% Called before each end_per_group.
- pre_end_per_group(Suite,Group,Config,State) ->
-     {Config, State}.
-
- %% Called after each end_per_group.
- post_end_per_group(Suite,Group,Config,Return,State) ->
-     {Return, State}.
-
  %% Called before each init_per_testcase.
- pre_init_per_testcase(Suite,TC,Config,State) ->
-     {Config, State#state{ ts = now(), total = State#state.suite_total + 1 } }.
-
- %% Called after each init_per_testcase (immediately before the test case).
- post_init_per_testcase(Suite,TC,Config,Return,State) ->
-     {Return, State}
-
-%% Called before each end_per_testcase (immediately after the test case).
- pre_end_per_testcase(Suite,TC,Config,State) ->
-     {Config, State}.
+ pre_init_per_testcase(_Suite,_TC,Config,State) ->
+     Now = erlang:monotonic_time(microsecond),
+     {Config, State#state{ ts = Now, suite_total = State#state.suite_total + 1 } }.
 
  %% Called after each end_per_testcase.
- post_end_per_testcase(Suite,TC,Config,Return,State) ->
-     TCInfo = {testcase, Suite, TC, Return, timer:now_diff(now(), State#state.ts)},
+ post_end_per_testcase(Suite,TC,_Config,Return,State) ->
+     Now = erlang:monotonic_time(microsecond),
+     TCInfo = {testcase, Suite, TC, Return, Now - State#state.ts},
      {Return, State#state{ ts = undefined, tcs = [TCInfo | State#state.tcs] } }.
 
- %% Called after post_init_per_suite, post_end_per_suite, post_init_per_group,
- %% post_end_per_group and post_end_per_testcase if the suite, group or test case failed.
- on_tc_fail(Suite, TC, Reason, State) ->
-     State.
-
  %% Called when a test case is skipped by either user action
  %% or due to an init function failing.  
- on_tc_skip(Suite, TC, Reason, State) ->
-     State.
+ on_tc_skip(_Suite, _TC, _Reason, State) ->
+     State#state{ skipped = State#state.skipped + 1 }.
 
- %% Called when the scope of the CTH is done
+ %% Called when the scope of the CTH is done.
  terminate(State) ->
-     io:format(State#state.file_handle, "~p.~n",
-                [{test_run, State#state.total, State#state.data}]),
-     file:close(State#state.file_handle),
-     ok.</code>
+     %% use append to avoid data loss if the path is reused
+     {ok, File} = file:open(State#state.filename, [write, append]),
+     io:format(File, "~p.~n", [results(State)]),
+     file:close(File),
+     ok.
+
+ results(State) ->
```

Comment:
> I left the extra space at the beginning to try and make sure the diff for my changes is a as legible as possible, showing only the meaningful changes. Would you like me to in this PR also get rid of that first extra column of space characters? (Maybe the diff preservation is not as interesting now that we are discussing bigger changes anyway)

Please remove the extra space in the example code.  We will see how it behaves ... it looks artificial, maybe there is some reason I'm missing ... let us check and hope it will be fine:-)

Reply:
👍